collect.py

```diff
@@ -119,25 +133,21 @@ def do_something(driver, elem_attributes=None):
             if not child.is_displayed() or not child.is_enabled():
                 continue
 
-            elem = child
-            break
+            res = get_elements_with_attributes(driver, elem_attributes, children)  # returns list of elements with attributes = child_attributes
+            if len(res) == 1:
+                elem = child
+                break
+            else:
+                # remove all elements with the same attributes
+                print('Found elements with same attributes')
+                # print(res)
+
+                list_ignore.extend(res)  # add elements to list_ignore
     else:
         if 'id' not in elem_attributes.keys():
-            body = driver.find_elements_by_tag_name('body')
-            assert len(body) == 1
-            body = body[0]
-
-            buttons = body.find_elements_by_tag_name('button')
-            links = body.find_elements_by_tag_name('a')
-            inputs = body.find_elements_by_tag_name('input')
-            selects = body.find_elements_by_tag_name('select')
-            children = buttons + links + inputs + selects
-
-            for child in children:
-                # Get all the attributes of the child.
-                if elem_attributes == get_all_attributes(driver, child):
-                    elem = child
-                    break
+            res = get_elements_with_attributes(driver, elem_attributes, children)  # returns elem if unique
```

Comment:
Remove the comment as it is self-explanatory and rename `res` with a better name.